src/harness/sas_test_harness.py

```diff
@@ -181,47 +192,58 @@ def __writeDumpFile(self, file_name, data):
+       These files will be served by SAS Test Harness.
     Args:
-       dump_records_list: List of dump_records object required to be written on the dump path.
+       dump_records_list: List of dump_records object required to 
+            be written on the dump path. Contains dump records in the format of 
+            [[cbsd1,..,cbsdn],[cbsda,..,cbsdc],[pp1,..,ppa3],[esc1,..,esc3]]
+            In this case four dump files will be created 
+            activity_dump_file_cbsd0.json - will contrain 1-n cbsd records
+            activity_dump_file_cbsd1.json - will contrain a-c cbsd records
+            activity_dump_file_zone2.json - will contrain 1-3 ppa records
+            activity_dump_file_esc_sensor3.json - will contrain 1-3 esc records
 
     Returns:
-       It returns None and all records are dumped in specified path.
-    """
+       It returns None and all records are dumped in specified path."""
+    
     fad_record_list = []
-    generated_file_name = None
-    record_url = None
-    dump_path = self.getDumpFilePath()
-
-    for dump_records in dump_records_list:
-      for index, dump_record in enumerate(dump_records):
-        if dump_record.has_key('id'):
-          if 'cbsd' in dump_record['id']:
-            generated_file_name = '{}.json'.format(dump_record['id'].decode('utf-8').split('/')[-1])
-            record_url = self.getBaseUrl() + '/cbsd/%s' % quote(dump_record['id'])
-          elif 'zone' in dump_record['id']:
-            generated_file_name = '{}.json'.format(dump_record['id']).split('/')[-1]
-            record_url = self.getBaseUrl() + '/' + dump_record['id']
-          elif 'esc_sensor' in dump_record['id']:
-            encoded_sensor_id = str(hashlib.sha1(dump_record['id']).hexdigest()).encode('utf-8')
-            generated_file_name = '{}.json'.format(encoded_sensor_id)
-            record_url = self.getBaseUrl() + '/%s/%s' % (dump_record['id'], encoded_sensor_id)
-          else:
-            raise Exception("ConfigurationError:incorrect record type. Unable to write FAD record")
-        else:
-          raise Exception("ConfigurationError:field id is not found. Unable to write FAD record")
-        if record_url is not None and generated_file_name is not None:
-          self.__writeDumpFile(generated_file_name, dump_record)
-          fad_record = self.__createFadRecord(record_url, dump_record)
-          fad_record_list.append(fad_record)
-        else:
-          raise Exception("SASTestHarnessError:Unable to create record url")
-      fad_file_name = 'FAD.json'
-      full_activity_dump = self.__createFadObject(fad_record_list)
-      self.__writeDumpFile(fad_file_name, full_activity_dump)
-      logging.info("FAD Records generated successfully."
-                   "The files are placed under the path:%s", self.getDumpFilePath())
+    dump_body = {}
+
+    for index, dump_records in enumerate(dump_records_list):
+        if len(dump_records) != 0:
```

Comment:
Try the following format to improve readability:

```
if len(dump_records) == 0:
  raise Exception("ConfigurationError:No records are configured")

...

if 'id' not in dump_record:
  # raise exception

# rest of your code here
```

Otherwise it's hard to tell which 'else' belongs to which 'if' and you end up with a lot of unnecessary indentation.

Reply:
Updated accordingly